pkg/cloud/gcp/actuators/machine/reconciler.go

```diff
@@ -164,3 +165,46 @@ func validateMachine(machine machinev1.Machine, providerSpec v1beta1.GCPMachineP
 	// Default values can also be set here
 	return nil
 }
+
+// Returns true if machine exists.
+func (r *Reconciler) instanceExists() (bool, error) {
+	if err := validateMachine(*r.machine, *r.providerSpec); err != nil {
+		return false, fmt.Errorf("failed validating machine provider spec: %v", err)
+	}
+	zone := r.providerSpec.Zone
+	// Need to verify that our project/zone exists before checking machine, as
+	// invalid project/zone produces same 404 error as no machine.
+	err := r.validateZone()
+	if err != nil {
```

Comment:
To be consistent with the call to `validateMachine`, let's continue with:

```go
if err := r.validateZone(); err != nil {
```

Ditto for the call to `r.computeService.InstanceGet` a few lines below.

Reply:
can't do this for r.computeSevice.InstancesGet if I implement you other suggest as err is not defined.

Reply:
OK - this instead? (excuse the duplicate comment elsewhere - should have been here): 

```go	
    if _, err := r.computeservice.instancesget(r.projectid, zone, r.machine.name); err != nil {
		if isNotFoundError(err) {
			return false, nil
		}
		klog.Errorf("Error getting running instances: %v", err)
		return false, err
	}
	return true, nil